src/ol/style/expressions.js

```diff
@@ -784,3 +785,58 @@ Operators['case'] = {
     return result;
   },
 };
+Operators['in'] = {
+  getReturnType: function (args) {
+    return ValueTypes.BOOLEAN;
+  },
+  toGlsl: function (context, args) {
+    assertArgsCount(args, 2);
+
+    const haystack = [];
+    const needle = [];
+    let result;
+    if (getValueType(args[1]) & ValueTypes.NUMBER_ARRAY) {
+      const keyword = expressionToGlsl(context, args[0]);
+      const input = args[1];
+      assertNumbers(input);
+      let i = 0;
+      let inputArr = ['array'];
+      // @ts-ignore
```

Comment:
```suggestion
      const input = /** @type {Array<number>} */ (args[1]);
      let i = 0;
      /** @type {Array<string|number>} */
      let inputArr = ['array'];
```
You can typecast the variables instead of using ts-ignore.
The assert here is unnecessary because the getValueType function already checks the type.